WalletWasabi/CoinJoin/Coordinator/Coordinator.cs

```diff
@@ -221,23 +221,46 @@ public async Task MakeSureTwoRunningRoundsAsync(Money? feePerInputs = null, Mone
 
 				int confirmationTarget = await AdjustConfirmationTargetAsync(lockCoinJoins: true).ConfigureAwait(false);
 
+				var configuredTimeout = TimeSpan.FromSeconds(RoundConfig.InputRegistrationTimeout);
 				if (runningRoundCount == 0)
 				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate, configuredTimeout);
 					round.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
 					round.StatusChanged += Round_StatusChangedAsync;
 					await round.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
 					Rounds.Add(round);
 
-					var round2 = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					var round2 = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate, 2 * configuredTimeout);
 					round2.StatusChanged += Round_StatusChangedAsync;
 					round2.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
 					await round2.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
 					Rounds.Add(round2);
 				}
 				else if (runningRoundCount == 1)
 				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					// If only one round is running, get the remaining time
+					var remainingTimeout = Rounds.FirstOrDefault(x => x.Status == CoordinatorRoundStatus.Running)?.RemainingInputRegistrationTime ?? TimeSpan.Zero;
```

Comment:
Consider adding another condition, that checks if the Round is in InputRegistration Phase. If not then TimeSpan.Zero should be returned to Create a round with the normal timeout (First condition below).

Reply:
RemainingInputRegistrationTime can be negative, which can destroy the calculations below. Make sure that it cannot be less than Zero, here or in CoordinatorRound.RemainingInputRegistrationTime.

Reply:
Done